File: src/testsettings/TestSettings.java
package testsettings;

import java.util.NoSuchElementException;
import java.util.Scanner;

/**
 * Created by colander on 22.3.17.
 * A general settings container.
 */
public class TestSettings {
    public static int SPECIES_RESET_COUNTER;
    public static int OUTPUT_FUNCTION;
    public static int ANN_FUNCTION;
    public static boolean SIGMOID_SHIFTED;
    public static boolean CONVERT_ANGLES;
    public static boolean NORMALIZE;
    public static double WEIGHT_MULTIPLIER;
    public static boolean PI_RANGE;
    public static float FRICTION;

    public static void set(String serialized) {
        Scanner sc = new Scanner(serialized);
        try {
            SPECIES_RESET_COUNTER = sc.nextInt();
            OUTPUT_FUNCTION = sc.nextInt();
            ANN_FUNCTION = sc.nextInt();
            SIGMOID_SHIFTED = sc.nextBoolean();
            CONVERT_ANGLES = sc.nextBoolean();
            NORMALIZE = sc.nextBoolean();
            WEIGHT_MULTIPLIER = sc.nextDouble();
            PI_RANGE = sc.nextBoolean();
            FRICTION = sc.nextFloat();
        } catch (NoSuchElementException e) {
            System.err.println("BROKEN CONFIG FILE");
        }
    }

    public static String serialize() {
        return SPECIES_RESET_COUNTER + "\n" + OUTPUT_FUNCTION + "\n" + ANN_FUNCTION + "\n" + SIGMOID_SHIFTED + "\n" + CONVERT_ANGLES + "\n" + NORMALIZE + "\n" + WEIGHT_MULTIPLIER + "\n" + PI_RANGE + "\n" + FRICTION + "\n";
    }
}

File: src/main/java/com/janboucek/crawler/settings/TestSettings.kt
package com.janboucek.crawler.settings

import java.util.*

/**
 * Created by colander on 22.3.17.
 * A general settings container.
 */
object TestSettings {
    var SPECIES_RESET_COUNTER = 0
    var OUTPUT_FUNCTION = 0
    var ANN_FUNCTION = 0
    var SIGMOID_SHIFTED = false
    var CONVERT_ANGLES = false
    var NORMALIZE = false
    var WEIGHT_MULTIPLIER = 0.0
    var PI_RANGE = false
    var FRICTION = 0f
    fun set(serialized: String) {
        val sc = Scanner(serialized)
        try {
            SPECIES_RESET_COUNTER = sc.nextInt()
            OUTPUT_FUNCTION = sc.nextInt()
            ANN_FUNCTION = sc.nextInt()
            SIGMOID_SHIFTED = sc.nextBoolean()
            CONVERT_ANGLES = sc.nextBoolean()
            NORMALIZE = sc.nextBoolean()
            WEIGHT_MULTIPLIER = sc.nextDouble()
            PI_RANGE = sc.nextBoolean()
            FRICTION = sc.nextFloat()
        } catch (e: NoSuchElementException) {
            throw IllegalStateException("BROKEN CONFIG FILE")
        }
    }

    fun serialize(): String {
        return """
            $SPECIES_RESET_COUNTER
            $OUTPUT_FUNCTION
            $ANN_FUNCTION
            $SIGMOID_SHIFTED
            $CONVERT_ANGLES
            $NORMALIZE
            $WEIGHT_MULTIPLIER
            $PI_RANGE
            $FRICTION
            
            """.trimIndent()
    }
}